DSharpPlus.Commands/Invocation/CommandEmitUtil.cs

```diff
@@ -19,106 +19,122 @@ internal static class CommandEmitUtil
     {
-        DynamicMethod dynamicMethod = new
-        (
-            $"{method.Name}-task-wrapper",
-            typeof(Task),
-            [typeof(object), typeof(object?[])]
-        );
-
-        ILGenerator il = dynamicMethod.GetILGenerator();
+        // Create the wrapper function
+        DynamicMethod dynamicMethod = new($"{method.Name}-task-wrapper", typeof(Task), [typeof(object), typeof(object?[])]);
 
-        EmitMethodWrapper(il, method);
+        // Create the wrapper logic
+        EmitMethodWrapper(dynamicMethod.GetILGenerator(), method);
 
+        // Return the delegate for the wrapper which invokes the method
         Func<object?, object?[], Task> taskWrapper = dynamicMethod.CreateDelegate<Func<object?, object?[], Task>>();
 
+        // Create an async wrapper around the task wrapper
         return async (object? instance, object?[] parameters) => await taskWrapper(instance, parameters);
     }
 
-    // the heavy lifting of emitting IL actually just works entirely the same for both, since we don't do
-    // anything further with the returned Task/ValueTask, and our wrapper is synchronous.
-    private static void EmitMethodWrapper(ILGenerator il, MethodInfo method)
+    /// <summary>
+    /// Writes the body of the wrapper function which invokes the command method.
+    /// </summary>
+    /// <param name="dynamicMethodIlGenerator"></param>
+    /// <param name="method"></param>
+    private static void EmitMethodWrapper(ILGenerator dynamicMethodIlGenerator, MethodInfo method)
     {
         ParameterInfo[] parameters = method.GetParameters();
-        ConstructorInfo argumentExceptionCtor = typeof(ArgumentException).GetConstructor([typeof(string), typeof(string)])!;
-
         if (!method.IsStatic)
         {
-            il.Emit(OpCodes.Ldarg_0);
-
-            // not static implies declared within a type
+            // Load the instance (this) onto the stack.
+            dynamicMethodIlGenerator.Emit(OpCodes.Ldarg_0);
             if (method.DeclaringType!.IsValueType)
             {
-                il.Emit(OpCodes.Unbox_Any, method.DeclaringType!);
+                // Change the handling behavior (unbox) of the instance (this) from a reference type to a value type.
+                dynamicMethodIlGenerator.Emit(OpCodes.Unbox_Any, method.DeclaringType);
             }
         }
 
+        // Load the arguments onto the stack.
         for (int i = 0; i < parameters.Length; i++)
         {
-            il.Emit(OpCodes.Ldarg_1);
-            il.Emit(OpCodes.Ldc_I4, i);
-            il.Emit(OpCodes.Ldelem_Ref);
+            // Ldarg_1 loads the array of arguments.
+            dynamicMethodIlGenerator.Emit(OpCodes.Ldarg_1);
+
+            // Ldc_I4 loads the index of the current argument.
+            dynamicMethodIlGenerator.Emit(OpCodes.Ldc_I4, i);
+
+            // Ldelem_Ref loads the element at the given index from the array.
+            dynamicMethodIlGenerator.Emit(OpCodes.Ldelem_Ref);
```

Comment:
if you insist on copying the instruction documentation, at least use the proper names - `ldarg.1`, `ldc.i4`, `ldelem.ref` - this goes for the entire file